src/components/TresCanvas.vue

```diff
@@ -107,17 +107,42 @@ const canvas = ref<HTMLCanvasElement>()
 */
 const scene = shallowRef<TresScene | Scene>(new Scene())
 
-const instance = getCurrentInstance()?.appContext.app
+const instance = getCurrentInstance()
 extend(THREE)
 
 const createInternalComponent = (context: TresContext, empty = false) =>
   defineComponent({
     setup() {
       const ctx = getCurrentInstance()?.appContext
-      if (ctx) { ctx.app = instance as App }
+      if (ctx) { ctx.app = instance?.appContext.app as App }
       provide('useTres', context)
       provide('extend', extend)
 
+      const provides = {}
```

Comment:
I think this is an improvement over the current situation where ancestors can't `provide` to `TresCanvas` components.

But it's got some limitations:

----

Because the solution here copies values, updates in ancestor `provide`s won't be reflected here unless:

* the provided value is non-primitive
* the provided value is updated via its keys

----

`provide`s in ancestors won't be `provide`d if they don't exist at the time of the instantiation of the `TresCanvas`.

----

Not sure if either of those are problems we want to address. But they should probably be documented if we don't address them.